glide-core/src/client/value_conversion.rs

```diff
@@ -379,9 +368,86 @@ pub(crate) fn convert_to_expected_type(
             )
                 .into()),
         },
+        // `FUNCTION LIST` returns an array of maps with nested list of maps.
+        // In RESP2 these maps are represented by arrays - we're going to convert them.
+        /* RESP2 response
+        1) 1) "library_name"
+           2) "mylib1"
+           3) "engine"
+           4) "LUA"
+           5) "functions"
+           6) 1) 1) "name"
+                 2) "myfunc1"
+                 3) "description"
+                 4) (nil)
+                 5) "flags"
+                 6) (empty array)
+              2) 1) "name"
+                 ...
+        2) 1) "library_name"
+           ...
+
+        RESP3 response
+        1) 1# "library_name" => "mylib1"
+           2# "engine" => "LUA"
+           3# "functions" =>
+              1) 1# "name" => "myfunc1"
+                 2# "description" => (nil)
+                 3# "flags" => (empty set)
+              2) 1# "name" => "myfunc2"
+                 ...
+        2) 1# "library_name" => "mylib2"
+           ...
+        */
+        ExpectedReturnType::ArrayOfMaps(type_of_map_values) => match value {
+            // empty array, or it is already contains a map (RESP3 response) - no conversion needed
+            Value::Array(ref array) if array.is_empty() || matches!(array[0], Value::Map(_)) => {
+                Ok(value)
+            }
+            // cluster (multi-node) response - go recursive
+            Value::Map(map) => convert_map_entries(
+                map,
+                Some(ExpectedReturnType::BulkString),
+                Some(ExpectedReturnType::ArrayOfMaps(type_of_map_values)),
+            ),
+            Value::Array(array) => convert_array_of_flat_maps(array, Some(*type_of_map_values)),
+            // Due to recursion, this will convert every map value, including simple strings, which we do nothing with
+            Value::BulkString(_) | Value::SimpleString(_) | Value::VerbatimString { .. } => {
+                Ok(value)
+            }
+            _ => Err((ErrorKind::TypeError, "Response couldn't be converted").into()),
+        },
+        // Not used for a command, but used as a helper for `ArrayOfMaps` to process the inner map.
+        // It may contain a string (name, description) or set (flags), or nil (description).
+        // The set is stored as array. See comment for `ArrayOfMapsRecursive`.
+        ExpectedReturnType::StringOrSet => match value {
+            Value::Nil => Ok(value),
+            Value::Array(_) => convert_to_expected_type(value, Some(ExpectedReturnType::Set)),
+            _ => convert_to_expected_type(value, Some(ExpectedReturnType::BulkString)),
```

Comment:
please keep the same format of matching specific value type and raising an error if unexpected type is received - don't use the default `_` option to convert all different types to BulkString as it might cause issues later on if new types will be introduced.

Reply:
Addressed in #1489